### DISCUSSION TITLE
Multi tenant schema on supabase

### DISCUSSION THREAD

**Discussion by christiankasongo** [supabase/supabase#16591] [Questions]
Hi, 
I am trying to achieve multi tenant on my project using supabase and Blazor wasm. I've created some tables on the public schema:

1. companies (store tenant details)
2. profile (creates a profile for auth user)
3. company_user ( links company and user)

I've also created some triggers:
- create schema for company - this runs when a new company has been added
- update_user_meta - this runs when a user has been linked to a company in the company_user table

I want user to only have access to data in the schema that belongs to their company.  example  company_A schema is called compA_db and it has tables such as task, stock, sales etc.. So if a user belongs to company a they should only be able to select, insert, update or delete on those tables in that schema.

Thank in advance

some resource I have used so far:
https://github.com/point-source/supabase-tenant-rbac
https://github.com/supabase-community/supabase-custom-claims
https://www.youtube.com/watch?v=IJoc6dKy03c&list=PL5S4mPUpp4OvS_lW6OGX2NTiQ_AHAQ3t-&ab_channel=Supabase

**Comment by GaryAustin1** [comment]
Nothing built in for sure.

https://github.com/orgs/supabase/discussions/11948 discusses using postgres roles for different classes of users.  In order to have entire schemas granted to certain users that is probably the only possible way.  The normal approaches using claims in the jwt will depend on RLS on individual tables, but you seem to be wanting the entire schema.

One issue would be if a user can belong to more than one role as their postgres role is based on their role in auth.users table in the linked discussion.

I imagine another approach to managing the role would be to have a server API or edge function create a jwt with the role claim based on the user's original jwt.  This is all presuming you are trying to use the REST API for the database which sets role based on the jwt.

**Reply by christiankasongo** [reply]
Thanks Gary
